google-cloud-spanner/src/main/java/com/google/cloud/spanner/CompositeTracer.java

```diff
@@ -0,0 +1,172 @@
+    this.children = ImmutableList.copyOf(children);
+  }
+
+  @Override
+  public Scope inScope() {
+    final List<Scope> childScopes = new ArrayList<>(children.size());
+
+    for (ApiTracer child : children) {
+      childScopes.add(child.inScope());
+    }
+
+    return new Scope() {
+      @Override
+      public void close() {
+        for (Scope childScope : childScopes) {
+          childScope.close();
+        }
+      }
+    };
+  }
+
+  @Override
+  public void operationSucceeded() {
+    for (ApiTracer child : children) {
+      child.operationSucceeded();
+    }
+  }
+
+  @Override
+  public void operationCancelled() {
+    for (ApiTracer child : children) {
+      child.operationCancelled();
+    }
+  }
+
+  @Override
+  public void operationFailed(Throwable error) {
+    for (ApiTracer child : children) {
+      child.operationFailed(error);
+    }
+  }
+
+  @Override
+  public void connectionSelected(String id) {
+    for (ApiTracer child : children) {
+      child.connectionSelected(id);
+    }
+  }
+
+  @Override
+  public void attemptStarted(int attemptNumber) {
+    for (ApiTracer child : children) {
+      child.attemptStarted(null, attemptNumber);
+    }
+  }
+
+  @Override
+  public void attemptStarted(Object request, int attemptNumber) {
+    for (ApiTracer child : children) {
+      child.attemptStarted(request, attemptNumber);
+    }
+  }
+
+  @Override
+  public void attemptSucceeded() {
+    for (ApiTracer child : children) {
+      child.attemptSucceeded();
+    }
+  }
+
+  @Override
+  public void attemptCancelled() {
+    for (ApiTracer child : children) {
+      child.attemptCancelled();
+    }
+  }
+
+  @Override
+  public void attemptFailed(Throwable error, Duration delay) {
+    for (ApiTracer child : children) {
+      child.attemptFailed(error, delay);
+    }
+  }
+
+  @Override
+  public void attemptFailedRetriesExhausted(Throwable error) {
+    for (ApiTracer child : children) {
+      child.attemptFailedRetriesExhausted(error);
+    }
+  }
+
+  @Override
+  public void attemptPermanentFailure(Throwable error) {
+    for (ApiTracer child : children) {
+      child.attemptPermanentFailure(error);
+    }
+  }
+
+  @Override
+  public void lroStartFailed(Throwable error) {
+    for (ApiTracer child : children) {
+      child.lroStartFailed(error);
+    }
+  }
+
+  @Override
+  public void lroStartSucceeded() {
+    for (ApiTracer child : children) {
+      child.lroStartSucceeded();
+    }
+  }
+
+  @Override
+  public void responseReceived() {
+    for (ApiTracer child : children) {
+      child.responseReceived();
+    }
+  }
+
+  @Override
+  public void requestSent() {
+    for (ApiTracer child : children) {
+      child.requestSent();
+    }
+  }
+
+  @Override
+  public void batchRequestSent(long elementCount, long requestSize) {
+    for (ApiTracer child : children) {
+      child.batchRequestSent(elementCount, requestSize);
+    }
+  }
+
+  public void addAttributes(String key, String value) {
```

Comment:
Where/when is this method called? It does not override any methods (at least it is not marked as such), and I can't find anything in this PR that seems to call it.

Reply:
This will be required for [MetricsTracer](https://github.com/googleapis/sdk-platform-java/blob/f952e3fc06a0920cd8bacc7fe030bd629433baae/gax-java/gax/src/main/java/com/google/api/gax/tracing/MetricsTracer.java#L232) 

Added in advance